network/slashing_violations_consumer.go

```diff
@@ -0,0 +1,51 @@
+package network
+
+import (
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// SlashingViolationsConsumer is a struct that logs a message for any slashable offences.
+// This struct will be updated in the future when slashing is implemented.
+type SlashingViolationsConsumer struct {
+	log zerolog.Logger
+}
+
+// NewSlashingViolationsConsumer returns a new SlashingViolationsConsumer
+func NewSlashingViolationsConsumer(log zerolog.Logger) *SlashingViolationsConsumer {
+	return &SlashingViolationsConsumer{log}
+}
+
+// OnUnAuthorizedSenderError logs a warning for unauthorized sender error
+func (c *SlashingViolationsConsumer) OnUnAuthorizedSenderError(identity *flow.Identity, peerID, msgType string, err error) {
+	c.log.Warn().
+		Err(err).
+		Str("peer_id", peerID).
+		Str("role", identity.Role.String()).
+		Str("peer_node_id", identity.NodeID.String()).
```

Comment:
please use `Hex("sender_id", logging.ID(identity.NodeID)).` instead of `Str` to log the node identifier. Also, please refactor it for subsequent methods.

Reply:
https://github.com/onflow/flow-go/pull/2648/commits/14c8ceebca0198b807e1ae408ff2e4fa185e941e